javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -47,6 +63,50 @@ module ExceptionXss {
     NotYetThrown() { this = "NotYetThrown" }
   }
 
+  // Consider using "if (err) {.. [do something with err] .. }" as an extra condition if there are too many FP's.
+  class Callback extends DataFlow::FunctionNode {
+    Callback() {
+      exists(DataFlow::CallNode call | call.getLastArgument().getAFunctionValue() = this) and
+      this.getNumParameter() = 2 and
+      this.getParameter(0).getName().regexpMatch("err.*") // Using "e" was considered. But that matches too many jQuery methods where "element" is shortened as "e".
```

Comment:
As mentioned offline, I think it is a slippery slope to rely on this naming heuristic.
Can we do something some semantic, like treating the immediate truthiness check for the first callback parameter in exactly the same way as the `catch` block in the previous iteration of this query?

```
function callback(e, a) {
    if (e) { // acts exactly like a catch block. We may even be able to require that this if-statement is the first non-trivial statement in the callback body.
        // handleError ...
       // early abort to ordinary or exceptional exit.
    }
    // ordinary logic
}
```

Additionally, I think we should look into supporting callback signatures with more than two parameters. But that can wait.